pyvisa-py/tcpip.py

```diff
@@ -381,13 +381,23 @@ def read(self, count):
             return (out + parts[0] + end_byte,
                     constants.StatusCode.success_termination_character_read)
 
+        # initial 'select_timout' is same as timeout or max 2s, so when no data arrived then max block time
+        select_timout = min(timeout, 2.0)
         while now - start <= timeout:
-            # use select to wait for read ready
-            select.select([self.interface], [], [])
-            last = read_fun(chunk_length)
+            # use select to wait for read ready, max `select_timout` seconds
+            try:
+                r, w, x = select.select([self.interface], [], [], select_timout)
+            except KeyboardInterrupt as err:
```

Comment:
What is the point of catching the exception to re-raise it directly here ?

Reply:
On windows select is no unblocked with KeyboardInterrupt but waits for timeout to be expired. But I tested it without catching the exception and it works OK, so removed try/except block.